src/lib/voiceSearchHelper/__tests__/index-test.ts

```diff
@@ -0,0 +1,170 @@
+import VoiceSearchHelper, { VoiceSearchHelperParams } from '..';
+
+const getHelper = (opts?: VoiceSearchHelperParams) =>
+  VoiceSearchHelper(
+    opts || {
+      searchAsYouSpeak: false,
+      onQueryChange: () => {},
+      onStateChange: () => {},
+    }
+  );
+
+type DummySpeechRecognition = () => void;
+declare global {
+  interface Window {
+    webkitSpeechRecognition?: SpeechRecognition | DummySpeechRecognition;
+    SpeechRecognition?: SpeechRecognition | DummySpeechRecognition;
+  }
+}
+
+describe('VoiceSearchHelper', () => {
+  beforeEach(() => {
+    delete window.webkitSpeechRecognition;
+    delete window.SpeechRecognition;
+  });
+
+  it('has initial state correctly', () => {
+    const helper = getHelper();
+    expect(helper.getState()).toEqual({
+      errorCode: undefined,
+      isSpeechFinal: undefined,
+      status: 'initial',
+      transcript: undefined,
+    });
+  });
+
+  it('is not supported', () => {
+    const helper = getHelper();
+    expect(helper.isBrowserSupported()).toBe(false);
+  });
+
+  it('is not listening', () => {
+    const helper = getHelper();
+    expect(helper.isListening()).toBe(false);
+  });
+
+  it('is supported with webkitSpeechRecognition', () => {
+    window.webkitSpeechRecognition = () => {};
+    const helper = getHelper();
+    expect(helper.isBrowserSupported()).toBe(true);
+  });
+
+  it('is supported with SpeechRecognition', () => {
+    window.SpeechRecognition = () => {};
+    const helper = getHelper();
+    expect(helper.isBrowserSupported()).toBe(true);
+  });
+
+  it('works with mock SpeechRecognition (searchAsYouSpeak:false)', () => {
+    let recognition;
+    window.SpeechRecognition = jest.fn().mockImplementation(() => ({
+      start() {
+        /* eslint-disable-next-line consistent-this */
```

Comment:
This doesn't seem to be necessary anymore.

Reply:
Without disabling the rule, it still gives me an error.

```
Unexpected alias 'recognition' for 'this'  consistent-this
```

Reply:
Indeed, my bad!